### ISSUE TITLE
xfce 4.18: item cannot be added to the panel: Plugin "xfce4-modifier-status-plugin" lacks a plugin register function. 

### ISSUE THREAD

**Issue by pinaceaepinus** [dwu/xfce4-modifier-status-plugin#1]
I'm trying this plugin on the upcoming debian bookworm which comes with xfce 4.18.

There, the item is in the item list (due to the *.desktop file I guess), but selecting it and clicking the "Add" button has no effect - nothing is added to the panel.

Following https://docs.xfce.org/xfce/xfce4-panel/debugging I start the panel with

```
  $ PANEL_DEBUG=1 xfce4-panel
```
...and when I hit "Add", this is log lines are generated:

```
...
xfce4-panel(external): register dbus path /org/xfce/Panel/Wrapper/7
xfce4-panel(module): new item (type=external-wrapper, name=xfce4-modifier-status-plugin, id=7)
xfce4-panel(external): xfce4-modifier-status-plugin-7: child spawned; pid=2083, argc=7
xfce4-panel(application): saving /panels/panel-1: ids=true, providers=false

(wrapper-2.0:2083): xfce4-panel-wrapper-CRITICAL **: 20:24:19.484: Plugin "xfce4-modifier-status-plugin" lacks a plugin register function.
xfce4-panel(external): xfce4-modifier-status-plugin-7: child exited with status 1280

(xfce4-panel:2016): xfce4-panel-WARNING **: 20:24:19.496: Plugin xfce4-modifier-status-plugin-7 exited with status 5, removing from panel configuration
xfce4-panel(external): xfce4-modifier-status-plugin-7: plugin unrealized; quitting child
xfce4-panel(application): saving /panels/panel-1: ids=true, providers=false
xfce4-panel(external): xfce4-modifier-status-plugin-7: plugin is being finalized
...
```

I guess `Plugin "xfce4-modifier-status-plugin" lacks a plugin register function.` is the important information here.

Any idea how to make this work with xfce 4.18? 

This plugin served me well on xfce 4.14 (Ubuntu 20.04/focal), thanks for that! :)

**Comment by dwu** [comment]
Thanks for letting me know. I wouldn't have noticed because I had originally ported this for a particular keyboard setup that I don't use anymore. I'll have a look at the panel changes in 4.18 and see whether I can come up with a solution.

**Comment by dwu** [comment]
@pinaceaepinus I've pushed a version with minor changes that builds and - after some very cursory testing - seems to run fine on a freshly updated Debian testing VM (as well as on Xubuntu 20.04... thanks to the Xfce guys for keeping things compatible).

It would be great if you could test if this fixes the issue for you as well.

**Comment by pinaceaepinus** [comment]
Work great, thanks!

I forgot to mention the xdt-csource change I also had to make.
Now I'm curious: How can the inclusion of <libxfce4panel/libxfce4panel.h> fix anything? Did it pull in a macro redefinition/overwrite?

**Comment by dwu** [comment]
Comparing 4.14 and 4.18 it seems that it was working previously only due to a long chain of includes. In 4.14 `xfce-panel-plugin.h` included `xfce-panel-macros-46.h` which included `xfce-panel-plugin-provider.h` which included `libxfce4panel.h` which then included `xfce-panel-macros.h` where `XFCE_PANEL_REGISTER_PLUGIN` is defined. In 4.18 this chain is no longer there.

The straight forward way would have been to include `libxfce4panel.h` right from the start. In fact, in this case including `xfce-panel-plugin.h` is not even required as it comes with `libxfce4panel.h`.

**Comment by dwu** [comment]
Fixed in 56f20e1b2062eb5e5c94a8bd030cb5ef99018b49

Please let me know if you find any further issues.
